engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnDecisionBehavior.java

```diff
@@ -70,27 +95,18 @@ public BpmnDecisionBehavior(
     // happens on element completion. We don't want to re-evaluate the decision for output mapping
     // related incidents.
     resultOrFailure.ifRight(
-        result ->
-            triggerProcessEventWithResultVariable(context, element.getResultVariable(), result));
+        result -> {
+          triggerProcessEventWithResultVariable(context, element.getResultVariable(), result);
+
+          final var decision = decisionOrFailure.get();
+          writeDecisionEvaluationEvent(decision, result, context);
+        });
```

Comment:
🔧 This change obfuscates what is meant by the comment block above. Please put writing the decisionEvaluationEvent into its own call to `resultOrFailure.ifRight()`. That way its more clear that the comment is only in relation to the triggerProcessEventWithResultVariable